superset-frontend/src/CRUD/CollectionTable.tsx

```diff
@@ -181,15 +207,75 @@ export default class CRUDCollection extends React.PureComponent<
     }));
   }
 
+  sortColumn(col: string, sort = 0) {
+    const { sortColumns } = this.props;
+    // default sort logic sorting string, boolean and number
+    const compareSort = (m: Sort, n: Sort) => {
+      if (typeof m === 'string') {
+        return (m || ' ').localeCompare(n);
+      }
+      return m - n;
+    };
+    return () => {
+      if (sortColumns?.includes(col)) {
+        // display in random order if no sort specified
```

Comment:
it's not a random order, this is the database column order(in other words, you can check this order in sqllab)